marklogic-data-hub-central/src/test/java/com/marklogic/hub/central/entities/search/impl/CreatedOnFacetHandlerTest.java

```diff
@@ -0,0 +1,41 @@
+package com.marklogic.hub.central.entities.search.impl;
+
+import com.marklogic.client.query.StructuredQueryBuilder;
+import com.marklogic.client.query.StructuredQueryDefinition;
+import com.marklogic.hub.central.entities.search.models.DocSearchQueryInfo;
+import com.marklogic.rest.util.Fragment;
+import org.jdom2.Namespace;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class CreatedOnFacetHandlerTest {
+
+    @Test
+    public void testCreatedOnDateRangeFacet() {
+        String constraintName = "createdOnRange";
+        String lowerBound = "2020-06-14";
+        String upperBound = "2020-06-17";
+        String expectedUpperBound = "2020-06-18";
+        DocSearchQueryInfo.FacetData facetData = new DocSearchQueryInfo.FacetData();
+        DocSearchQueryInfo.RangeValues rangeValues = new DocSearchQueryInfo.RangeValues();
+        rangeValues.setLowerBound(lowerBound);
+        rangeValues.setUpperBound(upperBound);
+        facetData.setRangeValues(rangeValues);
+        CreatedOnFacetHandler createdOnFacetHandler = new CreatedOnFacetHandler();
+        StructuredQueryBuilder queryBuilder = new StructuredQueryBuilder();
+
+        StructuredQueryDefinition queryDefinition = createdOnFacetHandler.buildQuery(facetData, queryBuilder);
+        Fragment fragment = new Fragment(queryDefinition.serialize(), Namespace.getNamespace("sch", "http://marklogic.com/appservices/search"));
```

Comment:
To avoid duplication, I recommend an AbstractFacetHandlerTest that has a "protected Fragment toFragment(StructuredQueryDefinition)" method which will construct the Fragment and initialize the needed Namespace.

Reply:
Good idea.